snoop/utils.py

```diff
@@ -1,5 +1,64 @@
 import subprocess
 import re
+import exifread
+from datetime import datetime
+
+def extract_gps_location(tags):
+    def ratio_to_float(ratio):
+        return float(ratio.num) / ratio.den
+
+    def convert(value):
+        d = ratio_to_float(value.values[0])
+        m = ratio_to_float(value.values[1])
+        s = ratio_to_float(value.values[2])
+        return d + (m / 60.0) + (s / 3600.0)
+
+    tags = {key: tags[key] for key in tags.keys() if key.startswith('GPS')}
+
+    lat = tags.get('GPS GPSLatitude')
+    lat_ref = tags.get('GPS GPSLatitudeRef')
+    lon = tags.get('GPS GPSLongitude')
+    lon_ref = tags.get('GPS GPSLongitudeRef')
+
+    if not lat or not lon or not lat_ref or not lon_ref:
+        return None
+
+    lat = convert(lat)
+    if lat_ref.values[0] != 'N':
+        lat = -lat
+    lon = convert(lon)
+    if lon_ref.values[0] != 'E':
+        lon = -lon
+    return "{}, {}".format(lat, lon)
+
+def convert_exif_date(str):
+    try:
+        date = datetime.strptime(str, "%Y:%m:%d %H:%M:%S")
+    except ValueError:
+        return None
+    return date.isoformat()
+
+def extract_exif(doc):
+    # detauls=False removes thumbnails and MakerNote (manufacturer specific
+    # information). See https://pypi.python.org/pypi/ExifRead#tag-descriptions
+    with doc.open(filesystem=True) as file:
```

Comment:
Hmm, it used to be a builtin on [python2](https://docs.python.org/2/library/functions.html#file), not any more. So never mind :)